File: sample/OpenCvSharp.Samples/Samples/SaveImage.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using OpenCvSharp;

namespace OpenCvSharp.Test
{
    /// <summary>
    /// Saving image using the format-specific save parameters
    /// </summary>
    class SaveImage
    {
        public SaveImage()
        {
            using (IplImage img = new IplImage(Const.Image16bit, LoadMode.Color))
            {
                // JPEG quality test
                img.SaveImage("q000.jpg", new JpegEncodingParam(0));
                img.SaveImage("q025.jpg", new JpegEncodingParam(25));
                img.SaveImage("q050.jpg", new JpegEncodingParam(50));
                img.SaveImage("q075.jpg", new JpegEncodingParam(75));
                img.SaveImage("q100.jpg", new JpegEncodingParam(100));

                using (IplImage q000 = new IplImage("q000.jpg", LoadMode.Color))
                using (IplImage q025 = new IplImage("q025.jpg", LoadMode.Color))
                using (IplImage q050 = new IplImage("q050.jpg", LoadMode.Color))
                using (IplImage q075 = new IplImage("q075.jpg", LoadMode.Color))
                using (IplImage q100 = new IplImage("q100.jpg", LoadMode.Color))
                using (CvWindow w000 = new CvWindow("quality 0", q000))
                using (CvWindow w025 = new CvWindow("quality 25", q025))
                using (CvWindow w050 = new CvWindow("quality 50", q050))
                using (CvWindow w075 = new CvWindow("quality 75", q075))
                using (CvWindow w100 = new CvWindow("quality 100", q100))
                {
                    Cv.WaitKey();
                }
            }
        }
    }
}

File: sample/CStyleSamplesVB/Samples/SaveImage.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports OpenCvSharp

' Namespace OpenCvSharpSamplesVB
Imports SampleBase

''' <summary>
''' Saving image using the format-specific save parameters
''' </summary>
Friend Module SaveImage
    Public Sub Start()
        Using img As New IplImage(FilePath.Image.Depth16Bit, LoadMode.Color)
            ' JPEG quality test
            img.SaveImage("q000.jpg", New JpegEncodingParam(0))
            img.SaveImage("q025.jpg", New JpegEncodingParam(25))
            img.SaveImage("q050.jpg", New JpegEncodingParam(50))
            img.SaveImage("q075.jpg", New JpegEncodingParam(75))
            img.SaveImage("q100.jpg", New JpegEncodingParam(100))

            Using q000 As New IplImage("q000.jpg", LoadMode.Color)
                Using q025 As New IplImage("q025.jpg", LoadMode.Color)
                    Using q050 As New IplImage("q050.jpg", LoadMode.Color)
                        Using q075 As New IplImage("q075.jpg", LoadMode.Color)
                            Using q100 As New IplImage("q100.jpg", LoadMode.Color)
                                Using w000 As New CvWindow("quality 0", q000)
                                    Using w025 As New CvWindow("quality 25", q025)
                                        Using w050 As New CvWindow("quality 50", q050)
                                            Using w075 As New CvWindow("quality 75", q075)
                                                Using w100 As New CvWindow("quality 100", q100)
                                                    Cv.WaitKey()
                                                End Using
                                            End Using
                                        End Using
                                    End Using
                                End Using
                            End Using
                        End Using
                    End Using
                End Using
            End Using
        End Using
    End Sub
End Module
' End Namespace
